src/overlays/actors/ovl_Bg_Vb_Sima/z_bg_vb_sima.c

```diff
@@ -21,13 +33,129 @@ const ActorInit Bg_Vb_Sima_InitVars = {
     (ActorFunc)BgVbSima_Update,
     (ActorFunc)BgVbSima_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Vb_Sima/BgVbSima_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Vb_Sima/BgVbSima_Destroy.s")
+static InitChainEntry sInitChain[] = { ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP) };
+
+Color_RGBA8_n colorYellow = { 0xFF, 0xFF, 0x00, 0xFF };
+Color_RGBA8_n colorRed = { 0xFF, 0x0A, 0x00, 0xFF };
+
+void BgVbSima_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgVbSima* this = THIS;
+    Actor* actorPtr = &this->dyna.actor;
```

Comment:
you must use `actorPtr` instead of `&this->dyna.actor` or `thisx` ?

Reply:
No, but I do need a pad of some sort. probably better to be honest and just go with a s32.